src/ol/source/tilejsonsource.js

```diff
@@ -44,21 +40,20 @@ ol.source.TileJSON = function(options) {
   });
 
   if (options.jsonp) {
-    var request = new goog.net.Jsonp(options.url);
-    request.send(undefined, this.handleTileJSONResponse.bind(this),
+    ol.net.jsonp(options.url, this.handleTileJSONResponse.bind(this),
         this.handleTileJSONError.bind(this));
   } else {
-    var xhr = new goog.net.XhrIo(new goog.net.CorsXmlHttpFactory());
-    goog.events.listen(xhr, goog.net.EventType.COMPLETE, function(e) {
-      if (xhr.isSuccess()) {
-        var response = /** @type {TileJSON} */(xhr.getResponseJson());
+    var xhr = new XMLHttpRequest();
+    xhr.open('GET', options.url, true);
+    xhr.onload = function(e) {
+      if (xhr.status < 400) {
```

Comment:
For network issues or CORS problems, `status` will be `0`, and we'd want the error handler to be called.  I'd change this to `xhr.status >= 200 && xhr.status < 300` (we aren't going to have the expected response text if a redirect isn't followed).